Rules and regulations (#4)
P3-MINI/csharp-site: content/regulations.en.md

- Project has a 2-week completion period from the time of assignment.
- Upon the instructor's request, students may be asked to present their submitted solutions during classes.
- The entire solution must be created independently.

Laboratory Tasks:

- Students can earn 0 to 12 points per laboratory task.
- Each laboratory task duration is 90 minutes during classes.
- The entire solution must be created independently.
- Laboratory tasks are divided into stages. Points for each stage are specified in the task description. Stages must be completed in the designated order.
- Each stage is checked and graded during classes.
- An excused absence allows solving the laboratory task at home within a week, with conversion points awarded at the end of the semester. Conversion points are calculated based on the average score of other students for that task (S1), the student's average score for other tasks (S2), and the result of the laboratory task solved at home (S3). The formula for points is: (S1 + S2) / 2 * S3 / 12.
- Laboratory tasks are done using the Arch Linux or Windows system on laboratory computers.

## 6. Permissible Aids During Verification of Learning Outcomes

> During the completion of project and laboratory tasks, students may use their own materials, system documentation, and the Internet, provided that the work evaluated is self-constructed. Particularly, using AI-generated solutions, such as chat GPT, Github Copilot, Google Gemini, and similar tools, is prohibited.

Author: ```suggestion
During the completion of project and laboratory tasks, students may use their own materials, system documentation, and the Internet, provided that the work evaluated is self-constructed. Particularly, using AI-generated solutions, such as ChatGPT, Github Copilot, Google Gemini, and similar tools, is prohibited.
```

During the completion of project and laboratory tasks, students may use their own materials, system documentation, and the Internet, provided that the work evaluated is self-constructed. Particularly, using AI-generated solutions, such as ChatGPT, GitHub Copilot, Google Gemini, and similar tools, is prohibited.

Any publicly available code fragments used must not constitute a significant portion of the solution and must be clearly marked with references to their source. Materials sourced from the Internet must be understandable to the student, and the student should be able to explain the operation of the copied code.

## 7. Course Completion and Final Grade Issuance

To pass, a student must earn a total of more than 50 points.

Grading scale:

\[0, 50\] points - 2.0 (fail)  
\(50, 60\] points - 3.0 (satisfactory)  
\(60, 70\] points - 3.5 (fairly good)  
\(70, 80\] points - 4.0 (good)  
\(80, 90\] points - 4.5 (above good)  
\(90, 100\] points - 5.0 (very good)

## 8. Grade Announcement Timeline and Method

Laboratory tasks and workshops are graded during the classes. Points are posted in the USOS system within 7 days after the classes.

> Workshops and projects are graded by the course tutor within 7 days of receiving the solution. In this case, points are posted in the USOS system no later than 7 days after grading.

Author: ```suggestion
Projects are graded by the course tutor within 7 days of receiving the solution. In this case, points are posted in the USOS system no later than 7 days after grading.
```

---
title: "Regulations"
---

> ## 1. Purpose of the Subject

Reviewer: Jedyna rzecz, jaką zauważyłem dotyczy formatowania - tytuły poszczególnych sekcji w wersji angielskiej mają "więcej" dużych liter niż ich odpowiedniki w wersji polskiej, czyli może:
```suggestion
## 1. Purpose of the subject
```
bo w wersji polskiej jest:
```
## 1. Przeznaczenie przedmiotu
```
a nie 
```
## 1. Przeznaczenie Przedmiotu
```

Author: A to tak ma być: https://www.grammarly.com/blog/punctuation-capitalization/capitalization-in-titles/
W angielskim tak się robi dla tytułów/nagłówków

- Workshops are done using the Arch Linux or Windows system on laboratory computers.

### Projects

- Students can earn 0 to 8 points per project.
- Project has a 2-week completion period from the time of assignment.
- Upon the instructor's request, students may be asked to present their submitted solutions during classes.
- The entire solution must be created independently.

### Laboratory Tasks

- Students can earn 0 to 12 points per laboratory task.
- Each laboratory task duration is 90 minutes during classes.
- The entire solution must be created independently.
- Laboratory tasks are divided into stages. Points for each stage are specified in the task description. Stages must be completed in the designated order.
- Each stage is checked and graded during classes.
- If a student has an excused absence, they may solve the laboratory task at home within a week. In this case, conversion points are awarded at the end of the semester.
  - Conversion points are calculated using the following method: S1 is the average score of other students for that task, S2 is the student's average score for other tasks, and S3 is the result of the laboratory task solved at home.

>   - The formula for points is: (S1 + S2) / 2 * S3 / 12.

Reviewer: Przypomniało mi się, że ta templatka, której używamy ma możliwość ładniejszego wyświetlania wzorów matematycznych: https://hugo-book-demo.netlify.app/docs/shortcodes/katex/
Możliwe, że tak zapisany będzie łatwiejszy do odczytania:
```tex
{{< katex display=true >}}
\frac{(S1 + S2)}{2} \cdot \frac{S3}{12}
{{< /katex >}}
```
